<a href="https://colab.research.google.com/github/tsvoronos/API202-students/blob/main/API_202_ReviewSession4_MD_Answers.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# API-202 Review Session #4
**Friday, February 17**

TF: Matthew Dodier

# Table of Contents
1. [Lecture Recap](#Lecture-Recap)
2. [Exercises](#Exercises)

# Lecture Recap

Our toolkit
* Thus far we have encountered a variety of tools to model relationships in the world through multiple regression.


| We are interested in… 	| Example 	| Tool 	|
|---	|---	|---	|
| ...the association between an outcome  and a binary explanatory variable 	| Do those residing in rural areas have  better/worse nutritional outcomes than  those residing in urban areas? 	| Dummy variables 	|
| ...the association between a binary outcome  variable and some explanatory variable 	| How does age relate to the likelihood that a person votes? 	| Linear probability model 	|
| ...how a relationship between two variables  is affected by a third variable 	| Is the relationship between health spending and health outcomes different for countries with nationalized health systems? 	| Interactions 	|
| ...whether a relationships is linear  or has "curvature" 	| Does income tend to increase with age in a constant way, or does it diminish over time 	| Quadratics, Logs 	|

* Today we will discuss investigating whether a relationship is linear or has curvature using quadratics and logs.

## Quadratic Transformations

Introduction to Quadratics

* Think back to our California class size, income and test score example. Consider the multivariate regression we ran:
$$
\text { testscr }=\beta_0+\beta_1 \text { str }+\beta_2 \text { income }+\varepsilon
$$
* Interpretation: an additional $\$ 1,000$ in income is associated with $\beta_2$ change in test scores...
  * Regardless of whether that family is poor or rich.
* Non-linear regression functions such as quadratics allow the predicted change in $Y$ associated with a change in $X$ to vary with $X$.

Often, linearity assumption doesn’t make sense
* So far, we have assumed a linear relationship between $X$ and $Y$
  * Non-linearities in regression
    * We can use non-linear transformations to allow the effect of $X$ on $Y$ to vary with $X$ 
    * e.g. $X^2$ when $X=$ years of education and $Y=$ earnings allows for increasing returns



Quadratics: Example

* Let's go back to our California school districts. Consider the relationship between average test scores and the average income (in thousands of dollars) across districts:

* A linear regression model says a one-unit change in income has the same marginal relationship at every level of income. Does that seem right - in the data and else?

* For example, consider: is a $\$1,000$ increase in income associated with the same increase in test scores if the average district income is $\$10,000$ and  $\$40,000$?

* Since the relationship between scores and income does not appear to be linear, we can model a non-linear relationship by adding a quadratic term:

$$
\text { TestScore }=\beta_0+\beta_1 \text { income }+\beta_2 \text { income }^2+\varepsilon
$$

Assume we estimate the following coefficients:
$$
\text { TestScore }=607.3+3.9 \text { income }-0.04 \text { income }{ }^2
$$

* We can't directly interpret the coefficient on income directly, since the slope is different depending on the income level you pick.

* To see how predicted test scores change with a one-unit $(\$ 1,000)$ change in income, we can no longer use the coefficient on the linear income term alone
We need to specify a particular level of income because now the relationship between $X$ and $Y$ depends on the level of $X$.

* We must calculate the change in $Y$ associated with a change in $X$ at each level of $X$ which we have an interest. Fill out the following table:  

| Income 	| $$ \text { testscore }=607.3+3.9 \text { income }-0.04 \text { income }{ }^2 $$ 	| Change in testscore 	|
|--------	|---------------------------------------------------------------------------------	|---------------------	|
| 10     	| $607.3+3.9(10)-0.04(10)^2=642.3$                                                	| 3.06                	|
| 11     	| $607.3+3.9(11)-0.04(11)^2=645.36$                                               	| 3.06                	|
| 20     	| $607.3+3.9(20)-0.04(20)^2=669.3$                                                	| 2.26                	|
| 21     	| $607.3+3.9(21)-0.04(21)^2=671.56$                                               	| 2.26                	|
| 40     	| $607.3+3.9(40)-0.04(40)^2=699.3$                                                	| 0.66                	|
| 41     	| $607.3+3.9(41)-0.04(41)^2=699.96$                                               	| 0.66                	|

$$
\text { TestScore }=607.3+3.9 \text { income }-0.04 \text { income }{ }^2
$$

Is the marginal association of income with scores increasing or decreasing? How can you tell from the SRF?

| $\hat{\beta}_1$ 	| $\hat{\beta}_2$ 	| Returns                                                                   	|
|-----------------	|-----------------	|---------------------------------------------------------------------------	|
| $+$             	| $+$             	| $Y$ **increases**/decreases with $X$ at an **increasing**/decreasing rate 	|
| $+$             	| $-$             	| $Y$ **increases**/decreases with $X$ at an increasing/**decreasing** rate 	|
| $-$             	| $-$             	| $Y$ increases/**decreases** with $X$ at an **increasing**/decreasing rate 	|
| $-$             	| $+$             	| $Y$ increases/**decreases** with $X$ at an increasing/**decreasing** rate 	|

## Logarithms

Introduction to Logs

* There can be various types of non-linearities in modeling the relationship between $Y$ and $\mathrm{X}$ - and different types of variables may require special such modeling.
* Think about the relationship between education and income. A small number of very wealthy individuals can make this relationship flatten out to the right.
  * A log-transformation of income can, in a certain sense, linearize it for us

* Key advantages to using logs
  * Coefficients have appealing interpretations (in percent terms), and we can therefore ignore the unit of measurements of our variables
  * Taking logs narrows the range of a variable, which usually makes estimates less sensitive to outliers
* Key limitations of logs:
  * By and large, cannot use them when the variable takes non-positive values
    * Logs are only defined for positive values (so 0's present a challenge)




Interpreting Logs

| Model       	| Regression Equation (PRF)                       	| Interpretation of $\beta_1$ (Math)                        	| Interpretation of $\beta_1$ (Words)                                              	|
|-------------	|-------------------------------------------------	|-----------------------------------------------------------	|----------------------------------------------------------------------------------	|
| Level-Level 	| $Y=\beta_0+\beta_1\left(X_1\right)+u$           	| $\Delta Y=\left(\beta_1\right) \Delta X_1$                	| A $1$ unit increase in $X_1$ is associated with a $\beta_1$ change in $Y$        	|
| Level-Log   	| $Y=\beta_0+\beta_1\left(\ln X_1\right)+u$       	| $\Delta Y=\left(\frac{\beta_1}{100}\right) \% \Delta X_1$ 	| A $1 \%$ increase in $X_1$ is associated with a $0.01 \beta_1$ change in $Y$     	|
| Log-Level   	| $\ln (Y)=\beta_0+\beta_1\left(X_1\right)+u$     	| $\% \Delta Y=\left(100 \beta_1\right) \Delta X_1$         	| A $1$ unit increase in $X_1$ is associated with a $100 \beta_1 \%$ change in $Y$ 	|
| Log-Log     	| $\ln (Y)=\beta_0+\beta_1\left(\ln X_1\right)+u$ 	| $\% \Delta Y=\left(\beta_1\right) \% \Delta X_1$          	| A $1 \%$ increase in $X_1$ is associated with a $\beta_1 \%$ change in $Y$       	|

**Case 1: Level-Log** ($X$ is in logs, $Y$ is not)
$$
Y=\beta_0+\beta_1 \ln (X)+u
$$

Interpretation:
A one-percent increase in $\mathrm{X}$ is associated with a $0.01 * \beta_1$ change in $\mathrm{Y}$
* The key assumption in the Level-Log model is that a $1 \%$ increase in $\mathrm{X}$ is associated with an increase in $\mathrm{Y}$ by a constant amount, $0.01^* \beta_1$, regardless of the level of $X$
* Whether that is a reasonable assumption will of course depend on the empirical context

Let's use an example relating $\mathrm{CO}_2$ emissions to GDP. The PRF is:
$$
\text { carbonemissions }_i=\beta_0+\beta_1 \ln \left(G D P_i\right)+u_i
$$

$$
\text { carbonemissions }_i=-70.72526+40.14255 \ln \left(G D P_i\right)+u_i
$$

Units in linear model:
* CO2 emissions, million tons
* GDP, billion USD

Question: How would you interpret the estimated coefficient on the variable $\ln \left(G D P_i\right)$?

Your answer: 

A one-percent increase in GDP is associated with $0.40$ million tons more $\mathrm{CO} 2$

* Take the $\beta_1$ coef, multiply it by 0.01: 40*0.01=0.4; so the answer is 0.40 million tons.
* We know that a one-percent increase in X is associated with a 0.01 times $\beta_1$ change in Y. In this case, a one percent increase in GDP is associated with a 40 times 0.01=0.4 million tons more CO2
Follow up Q: 

Question: Going from GDP $=1000$ to $G D P=1001$, what is the predicted change in emissions?

Your answer:

An increase of $0.040$ million tons $\mathrm{CO} 2$

* From 1000 to 1001 represents a 0.1 percent change, so 40*0.001 = 0.040M tonnes

**Case 2: Log-Level** ( $Y$ is in logs, $X$ is not)
$$
\ln (\mathrm{Y})=\beta_0+\beta_1 X+u
$$
Interpretation:
A one-unit increase in $\mathrm{X}$ is associated with a $100 * \beta_1$ percent change in $\mathrm{Y}$

* The key assumption in the Log-Level model is that a one-unit increase in $\mathrm{X}$ is associated with an increase in $\mathrm{Y}$ by a constant percent amount, $100 * \beta_1$, regardless of the level of $X$

* Aside: when $\beta_1$ suggests a large percent change (i.e. if $\beta_1>$ $0.25)$, this approximation is not very accurate. In those cases, use the exact formula: $\% \Delta \widehat{Y}=100\left[e^{\widehat{\beta_j}}-1\right]$

Let's return to the previous example relating $\mathrm{CO}_2$ emissions to GDP. The PRF is:
$$
\ln \left(\text {carbonemissions}_i\right)=\beta_0+\beta_1 G D P_i+u_i
$$

$$
\ln \left(\text {carbonemissions}_i\right)=1.533405 + .0043085 G D P_i+u_i
$$

Question: How would you interpret the estimated coefficient on the variable $g d p$ ?

a) A one percent increase in GDP is associated with $0.004$ percent more $\mathrm{CO} 2$

b) A one percent increase in GDP is associated with $0.4$ percent more $\mathrm{CO} 2$

c) A billion USD increase in GDP is associated with $0.004$ percent more $\mathrm{CO} 2$

d) A billion USD increase in GDP is associated with $0.4$ percent more $\mathrm{CO} 2$

Your answer:

d) A billion USD increase in GDP is associated with $0.4$ percent more $\mathrm{CO} 2$

**Case 3: Log-Log** (Both $X$ and $Y$ are in logs)
$$
\ln (Y)=\beta_0+\beta_1 \ln (X)+u
$$

Interpretation:
A one percent increase in $\mathrm{X}$ is associated with a $\beta_1$ percent change in $\mathrm{Y}$

In our data example:
$$
\ln \left(\text {carbonemissions}_i\right)=\beta_0+\beta_1 \ln \left(G D P_i\right)+u_i
$$

$$
\ln \left(\text {carbonemissions}_i\right)=-1.367377+1.065202 \ln \left(G D P_i\right)+u_i
$$

Question: How would you interpret the estimated coefficient on the variable $\ln \left(G D P_i\right)$?

Your answer:

A one-percent increase in GDP is associated with $1.06$ percent more CO2

Key Takeaways: Quadratics and Logs
* Non-linear regression functions allow us to model relationships that do not follow a traditional linear structure.
  * With quadratics and logs, linear regression can model curvature
  * Transformations allow you to model non-linear relationships using "linear" regression
* Using non-linear functions introduces important changes in the interpretation of our coefficients
* We can always understand how our dependent variable changes with a change in an independent variable by predicting values for with a given, then doing the same for, then taking the difference
* To determine when to use non-linear functions, consider the fit of your data, prior literature/practice on how to model particular variables, and theories as to what kind of relationships might exist

## Randomized Controlled Trials (RCTs)

RCTs: Rationale

* We will explore the strengths of RCTs and why they are considered the "gold standard" of study design.  RCTs allow us to **causally attribute** changes in outcome to an intervention by ensuring that **the relationship between the intervention** (the randomized treatment) **and any other determinant** of the outcome **equals zero**:

Remember $ \text {bias} = \gamma_1 \times \beta_2$

Under randomized treatment, $\gamma_1 = 0$ thus the bias equals 0

In its simplest form, an RCT works in the following way:

* Participants volunteer for the experiment. This constitutes the sample.
* The sample is ***randomly divided*** into “treatment” and “control” groups
  * Coin flip
  * Random number generator
  * Lottery
* “Treatment” group is offered the “treatment”
* “Control” group does not get the “treatment,” and may get a placebo.

Randomized Trials control for known + unknown

* Observational studies can suffer from bias, because people who take treatment differ from those who don't
* Some of these differences are observed but some are unobserved (e.g. motivation)
* By randomly assigning treatment, we control for observed + unobserved variation to get rigorous causal effects

Issue for causal inference
* There are 2 potential outcomes for every individual
  * 2 theoretical outcomes: *person with treatment* and *person without treatment*
  * we don’t get to observe both outcomes:
    * For an individual, we don't see both outcomes so we can't actually compare them

Selection bias
* The central problem of causal inference is figuring out how to **approximate the counterfactual of a treated group in the real world as well as possible.**
* For this to happen, we need our treatment and comparison group to be identical except for the fact that the treatment group received the intervention (e.g., college).
*How do RCTs help us approximate the treatment group's counterfactual?

RCTs & Selection bias
* Because members of the groups (treatment and control) do not differ systematically at the outset of the experiment, **any difference that subsequently arises between them can be attributed to the treatment** rather than to other factors (Michalopoulos, 2003).
* The distribution of the two groups on **all baseline characteristics** (observable and unobservable) will be identical up to a known degree of sampling error. One exception: participation in the program
* In the case of a properly conducted randomized trial, we can theoretically run a bivariate regression on a dummy variable for whether someone received the randomized treatment and recover the **true causal effect of the treatment:**
$$
Y=\beta_0+\beta_1 \text { Treat }+u
$$

To recap: RCTs help us estimate the counterfactual (with near identical treatment +
control groups)

* A well-designed RCT should produce two groups that look almost identical
* Same average potential outcome with treatment
* Same average potential outcome without treatment

In RCTs, avg treatment effect = difference in means

* The treatment and control groups have the same average potential outcomes for both scenarios
* So, the control group provides a valid counterfactual to the treatment group and vice versa
* Therefore, assuming perfect compliance:

Average Treatment Effect = Average Treated Outcome - Average Control Outcome

Estimate treatment effects with regression

* Recall that regression measures the average relationship between $X$ and $Y$
* So, for an RCT, we can alternatively measure ATE with regression: $\hat{Y}_i=\hat{\beta}_0+\hat{\beta}_1 \text{Treat}_i + \hat{u}_i$
* $\beta_0$ is the average outcome for the control group
* $\beta_0+\beta_1$ is avg outcome for treated group
* $\beta_1$ is the ATE

### Possible issues in RCTs

*Threat #1: Unbalanced Groups*
* If the treatment and controls group on average are different to begin with, then the groups are unbalanced
  * Some error in the implementation of the experiment?
  * Chance?
* Regardless, if the groups are unbalanced, the estimated treatment effect is likely to be biased

*Threat #2: Imperfect Compliance*
* In RCTs, we can randomly assign people to treatment / control
* However, it's often not possible to enforce whether or not treatment group individuals actually take treatment
* Also, often can't enforce that control individuals don't take treatment
* Imperfect compliance is when treatment group members don't get treatment or control group members do get it

*Threat #3: Attrition*
* Attrition (or non-response) means that some individuals disappear from the final sample (i.e. have no measured outcome)
* Whether this is a threat to internal validity depends on whether and how attrition is correlated with treatment
  * There is attrition bias if those who left the treatment group systematically differ from those who left the control group

*Threat #4: Spillover Effects*

* There are spillover effects if treating members of the treatment group affects the outcomes of the control group members
  * If there are positive spillover effects, such that the control group is positively affected, we will underestimate the true treatment effect
  * There could also be negative spillover effects, leading to overestimation, although this is typically less common

Key Takeaways: RCTs

* The counterfactual is at the core of causal inference. It is **what would have happened** to those receiving a treatment if they had not received that treatment.
* However, we cannot know the true counterfactual in the real world. Our primary challenge in trying to make causal claims lies in **approximating the counterfactual** of the treatment group with a well-selected control group.
* Randomized controlled trials allow us to approximate the counterfactual by ensuring that, on average, the control group and treatment group are **similar on all possible baseline variables** other than being offered the treatment.
* RCTs are considered to have high internal validity, though ensuring high external validity can be challenging.

# Exercises

In this part, you will explore how nonlinear varibles using data from the National Longitudinal Survey of Youth 2000 sample. The variables of interest in this exercise are: 

*   `income`: weekly earnings
*   `male`: a binary variable for whether the individual identifies as male
*   `yrs_educ`: years of education
*   `married`: a binary variable for whether the individual is married
*  `urban`: a binary variable for whether the individual is living in an urban (as opposed to rural) area
* `black`: a binary variable for whether the individual identifies as Black
* `hispanic`: a binary variable for whether the individual identifies as Hispanic
* `single`: a binary variable for whether the individual is single 
*   `age`: age in years.

In [2]:
#suppress warnings
options(warn =-1,dplyr.summarise.inform=FALSE)

library(tidyverse)
NLSY <- read.csv("https://github.com/tsvoronos/API202-students/blob/9b0d9d54fff8677151f5ae86df096fb55e6b39d7/data/nlsy2000_2012.csv?raw=1")

head(NLSY)


── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──

✔ ggplot2 3.4.1     ✔ purrr   1.0.1
✔ tibble  3.1.8     ✔ dplyr   1.1.0
✔ tidyr   1.3.0     ✔ stringr 1.4.1
✔ readr   2.1.4     ✔ forcats 1.0.0

── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()



age black hispanic income    single married yrs_educ urban male
1 21  0     0         870.1858 1      0        9       1     0   
2 18  0     0         755.9383 1      0       10       1     0   
3 23  0     0        1113.9143 1      0       13       1     1   
4 22  0     0         792.1168 0      1       12       1     1   
5 19  0     0         809.2539 1      0       12       1     1   
6 19  0     0         618.8412 1      0       13       1     0

**1. Transform income into the log of income.**



In [3]:
# Your code here

NLSY <- NLSY %>%
  mutate(log_income = log(income))

**2. Run a regression of log income on marital status, hispanic, and age. Interpret each of the three coefficients on married, hispanic, and age, and discuss whether each is statistically significant at the 5% level.**

In [4]:
# Your code here

reg1 <- lm(log_income ~ married + hispanic + age, data = NLSY)
summary(reg1)


Call:
lm(formula = log_income ~ married + hispanic + age, data = NLSY)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.09104 -0.20711 -0.04515  0.19780  1.23990 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)  5.435364   0.141170  38.502   <2e-16 ***
married     -0.014240   0.025002  -0.570    0.569    
hispanic    -0.020245   0.046150  -0.439    0.661    
age          0.063958   0.007013   9.120   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.338 on 996 degrees of freedom
Multiple R-squared:  0.08008,	Adjusted R-squared:  0.07731 
F-statistic:  28.9 on 3 and 996 DF,  p-value: < 2.2e-16


In [5]:
# Alternatively we can include the log transformation of income in our regression 
# using the I() function. The I() function changes the class of an object to 
# indicate that it should be treated ‘as is’.

reg1b <- lm(I(log(income)) ~ married + hispanic + age, data = NLSY)
summary(reg1b)


Call:
lm(formula = I(log(income)) ~ married + hispanic + age, data = NLSY)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.09104 -0.20711 -0.04515  0.19780  1.23990 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)  5.435364   0.141170  38.502   <2e-16 ***
married     -0.014240   0.025002  -0.570    0.569    
hispanic    -0.020245   0.046150  -0.439    0.661    
age          0.063958   0.007013   9.120   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.338 on 996 degrees of freedom
Multiple R-squared:  0.08008,	Adjusted R-squared:  0.07731 
F-statistic:  28.9 on 3 and 996 DF,  p-value: < 2.2e-16


Your answer here:

Married individuals have approximately $1.4 \%$ lower weekly earnings than single individuals controlling for ethnicity and age. This is not statistically significant at the $5 \%$ level.

Hispanic individuals have approximately $2.0 \%$ lower weekly earnings than non-hispanic individuals, controlling for marital status and age. This is not statistically significant at the $5 \%$ level.

An additional year of age is associated with approximately $6.4 \%$ higher earnings, controlling for marital status and ethnicity. This is statistically significant at the $5 \%$ level.


**3. Add a quadratic term for age to the regression in question 2.**

In [6]:
NLSY <- NLSY %>%
  mutate(age_sq = age*age)

reg2 <- lm(log_income ~ married + hispanic + age + age_sq, data = NLSY)
summary(reg2)


Call:
lm(formula = log_income ~ married + hispanic + age + age_sq, 
    data = NLSY)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.10346 -0.21188 -0.04686  0.19259  1.22607 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)  
(Intercept)  2.369914   1.535071   1.544   0.1229  
married     -0.014486   0.024964  -0.580   0.5619  
hispanic    -0.016823   0.046112  -0.365   0.7153  
age          0.371409   0.153470   2.420   0.0157 *
age_sq      -0.007661   0.003820  -2.005   0.0452 *
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.3375 on 995 degrees of freedom
Multiple R-squared:  0.08378,	Adjusted R-squared:  0.0801 
F-statistic: 22.75 on 4 and 995 DF,  p-value: < 2.2e-16



Call:
lm(formula = I(log(income)) ~ married + hispanic + age + I(age^2), 
    data = NLSY)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.10346 -0.21188 -0.04686  0.19259  1.22607 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)  
(Intercept)  2.369914   1.535071   1.544   0.1229  
married     -0.014486   0.024964  -0.580   0.5619  
hispanic    -0.016823   0.046112  -0.365   0.7153  
age          0.371409   0.153470   2.420   0.0157 *
I(age^2)    -0.007661   0.003820  -2.005   0.0452 *
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.3375 on 995 degrees of freedom
Multiple R-squared:  0.08378,	Adjusted R-squared:  0.0801 
F-statistic: 22.75 on 4 and 995 DF,  p-value: < 2.2e-16


In [ ]:
# Alternatively we can include the log transformation of income and age squared
# in our regression using the I() function. The I() function changes the class of
# an object to indicate that it should be treated ‘as is’.

reg2b <- lm(I(log(income)) ~ married + hispanic + age + I(age^2), data = NLSY)
summary(reg2b)

**4. What’s the predicted difference in income between people with 17 and 18 years of age, according to the regression in question 3?**

Your answer here:

$(0.371409 \times 18 - 0.007661 \times 18^2) - (0.371409 \times 17 - 0.007661 \times 17^2) \approx 0.103$ 

People aged 18 are predicted to earn $10.3 \%$ more per week than people aged 17.

In [7]:
# We could also calculate this predicted difference in R using the following code:

predict(reg2,
newdata = tibble(age=c(17,18), age_sq=c(17^2,18^2), married=c(0,0), hispanic=c(0,0))) %>% diff()

2 
0.1032664

**5. What is the predicted difference in income between people with 20 and 21 years of age, according to the regression in question 3?**

Your answer here:

$(0.371409 \times 21 - 0.007661 \times 21^2) - (0.371409 \times 20 - 0.007661 \times 20^2) \approx 0.057$ 

People aged 21 are predicted to earn $5.7 \%$ more per week than people aged 20.

In [9]:
# We could also calculate this predicted difference in R using the following code:

predict(reg2,
newdata = tibble(age=c(20,21), age_sq=c(20^2,21^2), married=c(0,0), hispanic=c(0,0))) %>% diff()

2 
0.05729906